### ISSUE TITLE
Cannot deploy functions in parallel

### ISSUE THREAD

**Issue by riderx** [supabase/cli#431]
# Bug report

## Describe the bug

i have a script who upload all my cloud function after CI pass and it take around 2 min.
So i tryed to parallelize them and i got issue 
```
Error: Command failed: supabase functions deploy on_shared_created
Error: Error bundling function: fork/exec /home/runner/.supabase/deno: text file busy


    at ChildProcess.exithandler (node:child_process:400:12)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1093:16)
    at Socket.<anonymous> (node:internal/child_process:451:11)
    at Socket.emit (node:events:513:28)
    at Pipe.<anonymous> (node:net:7[57](https://github.com/Cap-go/capgo/runs/8101800008?check_suite_focus=true#step:10:58):14) {
  code: 1,
  killed: false,
  signal: null,
  cmd: 'supabase functions deploy on_shared_created',
  stdout: 'Bundling \x1B[1mon_shared_created\x1B[0m\n',
  stderr: 'Error: Error bundling function: fork/exec /home/runner/.supabase/deno: text file busy\n' +
    '\n'
}
Error: Process completed with exit code 1.
```

## To Reproduce

create a script who run mutiple function upload in parallel

## Expected behavior

should be ok

## System information

- OS: [e.g. macOS, Windows]
- Browser (if applies) [e.g. chrome, safari]
- Version of supabase-js: [e.g. 6.0.2]
- Version of Node.js: [e.g. 10.10.0]

**Comment by soedirgo** [comment]
Closing this in favor of #456
